service/pivot/handlers_pivot.go

```diff
@@ -87,7 +86,7 @@ func ModelAssertion(w http.ResponseWriter, r *http.Request) response.ErrorRespon
 	assertions := []asserts.Assertion{}
 
 	// Build the model assertion headers for the original model
-	assertionHeaders, keypair, err := assert.CreateModelAssertionHeaders(substore.FromModel)
+	assertionHeaders, keypair, err := datastore.CreateModelAssertionHeaders(substore.FromModel)
```

Comment:
if I undersand correctly this will continue to produce a new model for each pivoting device? in theory we should change this too to reuse the same model cached in the db if possible,  can be done later or not done given that pivot is a historical feature, nowadays we have remodel in principle. anyway wanting to make sure it's clear where we stand after this set of changes

Reply:
Yes, this is correct. I would change this in the next PR after we fixed all the problems with the model assertion